### ISSUE TITLE
[DOC] Data Prepper plugin naming consistency

### ISSUE THREAD

**Issue by dlvenable** [opensearch-project/documentation-website#3666]
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.

The new Data Prepper documentation has some inconsistencies in the Pipelines section.

Each pipeline component has a specific component name. Some examples are `otel_trace_source`, `s3`, `opensearch`, `grok`, `parse_json`. Data Prepper's convention is that they are all lowercase and use underscores to separate words. These identifiers are what pipeline authors use when creating pipelines.

The documentation uses different terms for what is the same identifier. 

Update the documentation to use the pipeline component name consistently. This should be reflected in the URL path, the table of contents, descriptions, etc.

### Example 1 - otel_trace_source

The documentation currently has:

* URL: https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sources/otel-trace/
* Title in table-of-contents: "otel_trace_source source"
* Title on the page: "otel_trace source"
* Intro paragraph: "The `otel_trace` source is a source for the OpenTelemetry Collector. The following table describes options you can use to configure the `otel_trace` source."

The actual name in Data Prepper is "otel_trace_source".

An example pipeline looks like:

```
source:
  otel_trace_source:
```

The documentation should be updated to use:

* URL: https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sources/otel_trace_source/
* Title in table-of-contents: "otel_trace_source"
* Title on the page: "otel_trace_source"
* Intro paragraph: "The `otel_trace_source` is a source for the OpenTelemetry Collector. The following table describes options you can use to configure the `otel_trace_source`."

### Example 2 - parse_json

The documentation currently has:

* URL https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/processors/parse-json/
* Title in table-of-contents: "Parse JSON"
* Title on the page: "Parse JSON"
* Intro paragraph: "The `parse_json` processor parses JSON data for an event, including any nested fields. The following table describes several optional parameters you can configure in the `parse_json` processor."

The actual name in Data Prepper is "otel_trace_source".

An example pipeline looks like:

```
processor:
  parse_json:
```

The documentation should be updated to use:

* URL https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/processors/parse_json/
* Title in table-of-contents: "parse_json"
* Title on the page: "parse_json"
* Intro paragraph: (No change needed here; it is already correct)

**Comment by dlvenable** [comment]
I think every URL is currently using `-` which is the documentation convention. I think for these processors, we should use `_` to stay consistent with the Data Prepper naming convention.

So almost every component will need the URL updated.

**Comment by Naarcha-AWS** [comment]
@dlvenable: If we update the URLs we'll have to add an additional redirect. Is that ok?

**Comment by dlvenable** [comment]
Yes, I think it is ok to have the redirection.

**Comment by hdhalter** [comment]
@dlvenable - Let's keep as-is for now. This is a major change for us to make to all the data prepper files making them inconsistent with our standard naming convention. In addition, updating the URLs will have an impact on the analytics we are tracking. Perhaps we can revisit this once we de-couple the content from the OpenSearch versioning or do major updates to the content that affect the structure.

**Comment by dlvenable** [comment]
@hdhalter , I don't think we need to solve the URL issue right now. But, we should keep this in mind going forward. 

So for new plugins, we can use a simple heuristic. Take the plugin name and replace underscores with hyphens. So for example, `otel_trace_source` becomes `otel-trace-source.md`

**Comment by Naarcha-AWS** [comment]
One caveat to this, when searching using "List to map processor" would be better for SEO since people aren't likely to search with the underscores. Thoughts?

**Comment by vagimeli** [comment]
@Naarcha-AWS @dlvenable Here's an open source documentation source re: SEO. Passing along to help support us make informed decision-making https://www.writethedocs.org/guide/seo/.

**Comment by Naarcha-AWS** [comment]
@natebower: From an editorial perspective, what are your thoughts for the title in the TOC, "`parse_json` processor" versus "Parse JSON processor"?

**Comment by natebower** [comment]
Let's get the arguments tomorrow and decide on an approach.

**Comment by dlvenable** [comment]
Regarding SEO, we can use page metadata, correct?
